plugins/inputs/systemd_units/systemd_units_linux.go

```diff
@@ -158,12 +161,34 @@ func (s *SystemdUnits) Init() error {
 	}
 	s.filter = f
 
+	switch s.Scope {
+	case "", "system":
+		s.scope = "system"
+		s.user = ""
+	case "user":
+		s.scope = "user"
+		s.user = os.Getenv("USER")
+	default:
+		return fmt.Errorf("invalid 'scope' %q", s.Scope)
+	}
+
 	return nil
 }
 
 func (s *SystemdUnits) Start(telegraf.Accumulator) error {
+	var (
+		client *dbus.Conn
+		err    error
+	)
+
```

Comment:
Please declare the variables close to their use to avoid accidental reuse...

Reply:
ok